doc/eID/eid-integration-magnus.md

```diff
@@ -0,0 +1,37 @@
+# UseID Integration
+
+To integrate UseID mobile app and web widget, the following steps are necessary for services already having an existing eID integration.
+
+## Integration
+
+### 1. Embedding Widget
+
+eService embeds widget via JavaScript by adding the following empty container to the HTML page:
+
+```html
+<div id="useid-widget-container" data-tc-token-url="{}"></div>
+```
+
+and including this script in the page:
+
+```html
+<script async src="https://useid.ds4g.net/widget.js"></script>
+```
+
+As value for `data-tc-token-url` the already known tcTokenURL is used.
+
+> The eService usually presents a link to the user which points to the eID client application (e.g. AusweisApp2) including the tcTokenURL as a query parameter. With UseID, this is replaced by the web widget displaying a QR code which can be scanned to start the UseID mobile app.
+
+### 2. ID process
+
+The process of scanning the eID has not changed for the eService. Changes for the users are in using the UseID mobile app and their smartphone as scanning devices.
+
+### 3. Success and Refresh
+
+The user is returned to the `refreshAddress` of the eService, which is provided in the TC token, on the device displaying the widget.
+
+## Security
+
+Instead of staying on the same device, a QR code is displayed to be scanned by the user's smartphone. At success, the user returns to the page displaying the widget and gets forwarded to the refresh address. The embedded widget (locally in the browser) and the smartphone establish an end-to-end encrypted channel so that any communication going through UseID servers can only be interpreted by the user's devices; especially `tcTokenURL` and `refreshAddress` are never exposed to UseID servers.
```

Comment:
I am wondering, the QR Code would be still valid after successful identification via mobile app? should we also invalidate the QR code as well?

Reply:
Yes, I think so, too. We need to keep an eye on the handling of sessions, when to start and delete them, and then it should be fine.

Reply:
1. why does refresh address more crucial?
2. especially tcTokenURL and refreshAddress are never exposed to UseID servers. I am a bit confused by this sentence.

let's say, this is an `tcTokenUrl`
https://www.foo.bar/eid-server-demo-app/result/getTcToken.html;jsessionid=RANDOMSESSIONID and it will return 

```xml
<TCTokenType>
	  <ServerAddress>https://bar.foo.de:443</ServerAddress>
	  <SessionIdentifier>SESSIONIDENTIFIER</SessionIdentifier>
	  <RefreshAddress>https://www.foo.bar:443/eid-server-demo-app/result/response.html;jsessionid=RANDOMSESSIONID</RefreshAddress>
	  <Binding>urn:liberty:paos:2006-08</Binding>
	  <PathSecurity-Protocol>urn:ietf:rfc:4279</PathSecurity-Protocol>
	  <PathSecurity-Parameters>
	  	<PSK>FOOBAR</PSK>
	  </PathSecurity-Parameters>
</TCTokenType>
```

so it already includes RefreshAddress, why do we need both? I am a bit confused...

Reply:
What do you mean by "why do we need both"? Maybe we can have a quick chat on that, that's easier than writing?